app/uk/gov/hmrc/gform/formtemplate/RepeatingComponentService.scala

```diff
@@ -81,23 +81,42 @@ object RepeatingComponentService {
     }
 
     section.copy(
-      title = buildText(Some(section.title), index, section.fieldToTrack.get, data).getOrElse(""),
-      shortName = buildText(section.shortName, index, section.fieldToTrack.get, data),
+      title = buildText(Some(section.title), index, data).getOrElse(""),
+      shortName = buildText(section.shortName, index, data),
       fields = section.fields.map(copyField)
     )
   }
 
-  private def buildText(template: Option[String], index: Int, fieldToTrack: VariableInContext,
-    data: Map[String, String]): Option[String] = {
+  private def buildText(template: Option[String], index: Int, data: Map[String, String]): Option[String] = {
+
+    def evaluateTextExpression(str: String) = {
+      val field = str.replaceFirst("""\$\{""", "").replaceFirst("""\}""", "")
+      if (field.startsWith("n_")) {
+        if (index == 1) {
+          val fieldName = field.replaceFirst("n_", "")
+          data.getOrElse(fieldName, "")
+        } else {
+          val fieldName = field.replaceFirst("n_", s"${index - 1}_")
+          data.getOrElse(fieldName, "")
+        }
+      } else {
+        data.getOrElse(field, "")
+      }
+    }
 
-    val fieldName = if (index == 1) fieldToTrack.field else s"${index - 1}_${fieldToTrack.field}"
-    val textToInsert = data.getOrElse(fieldName, "")
+    def getEvaluatedText(str: String) = {
```

Comment:
How about moving this part and above to separate object (let's say `ExpressionEvaluator`)? How do we do expression evaluation in other places? Maybe what I'm saying here doesn't make sense.